src/Database/Relations/BelongsToMany.php

```diff
@@ -1,10 +1,147 @@
-<?php namespace Winter\Storm\Database\Relations;
+<?php
 
+namespace Winter\Storm\Database\Relations;
+
+use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\Relations\BelongsToMany as BelongsToManyBase;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Winter\Storm\Database\Pivot;
 
-class BelongsToMany extends BelongsToManyBase
+class BelongsToMany extends BelongsToManyBase implements Relation
 {
     use Concerns\BelongsOrMorphsToMany;
+    use Concerns\CanBeCounted;
+    use Concerns\CanBeDetachable;
+    use Concerns\CanBeExtended;
+    use Concerns\CanBePushed;
     use Concerns\DeferOneOrMany;
     use Concerns\DefinedConstraints;
+    use Concerns\HasRelationName;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(
+        Builder $query,
+        Model $parent,
+        $table,
+        $foreignPivotKey,
+        $relatedPivotKey,
+        $parentKey,
+        $relatedKey,
+        $relationName = null
+    ) {
+        parent::__construct($query, $parent, $table, $foreignPivotKey, $relatedPivotKey, $parentKey, $relatedKey, $relationName);
+        $this->extendableRelationConstruct();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getPivotClass()
+    {
+        return $this->using ?? Pivot::class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setSimpleValue($value): void
+    {
+        $relationModel = $this->getRelated();
+
+        /*
+         * Nulling the relationship
+         */
+        if (!$value) {
+            // Disassociate in memory immediately
+            $this->parent->setRelation($this->relationName, $relationModel->newCollection());
+
+            // Perform sync when the model is saved
+            $this->parent->bindEventOnce('model.afterSave', function () {
+                $this->detach();
+            });
+            return;
+        }
+
+        /*
+         * Convert models to keys
+         */
+        if ($value instanceof Model) {
+            $value = $value->getKey();
+        }
+        elseif (is_array($value)) {
```

Comment:
```suggestion
        } elseif (is_array($value)) {
```
We should probably make some adjustments to our published coding style and apply it in bulk across our codebases soon.

Reply:
We do cover it under **Curly braces for condition blocks** here: https://wintercms.com/docs/v1.2/docs/architecture/developer-guide#php-coding-standards

It's probable that I missed this or copied it from some old code.